File: VacationDistribution/VacationDistribution/ClientApp/app/Entities/Employee.ts
export enum EPosts {
    Dev = 1,
    QA = 2,
    TeamLead = 3
}

export const EPostsToLabelMapping = [
    { value: EPosts.Dev, name: 'Dev' },
    { value: EPosts.QA, name: 'QA' },
    { value: EPosts.TeamLead, name: 'TeamLead' }
];

export class Employee {
    constructor(
        public id?: number,
        public firstName?: string,
        public middleName?: string,
        public lastName?: string,
        public post?: EPosts
    ) {}
}

File: VacationDistribution/VacationDistribution/Models/Entities/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace VacationDistribution.Models
{
    public class Employee
    {
        /// <summary>
        /// Идентификатор сотрудника
        /// </summary>
        public int Id { get; set; }
        /// <summary>
        /// Имя
        /// </summary>
        public string FirstName { get; set; }
        /// <summary>
        /// Отчество
        /// </summary>
        public string MiddleName { get; set; }
        /// <summary>
        /// Фамилия
        /// </summary>
        public string LastName { get; set; }
        /// <summary>
        /// Должность
        /// </summary>
        public EPosts Post { get; set; }
    }

    public enum EPosts
    {
        Dev = 1,
        QA = 2,
        TeamLead = 3
    }
}
